powermon/outputs/__init__.py

```diff
@@ -0,0 +1,117 @@
+    if outputsConfig is None:
+        _outputs.append(parseOutputConfig({"type": "screen", "format": "simple"}))
+    elif isinstance(outputsConfig, str):
+        # eg 'screen'
+        _outputs.append(parseOutputConfig(outputsConfig))
+    elif isinstance(outputsConfig, list):
+        # eg [{'type': 'screen', 'format': 'simple'}, {'type': 'screen', 'format': {'type': 'htmltable'}}]
+        for outputConfig in outputsConfig:
+            _outputs.append(parseOutputConfig(outputConfig))
+    elif isinstance(outputsConfig, dict):
+        # eg {'format': 'table'}
+        _outputs.append(parseOutputConfig(outputsConfig))
+    else:
+        pass
+    return _outputs
+
+
+def parseOutputConfig(outputConfig, topic=None, schedule_name=None, device=None, mqtt_broker=None):
+    log.debug("parseOutputConfig, config: %s, topic: %s" % (outputConfig, topic))
+    topic_override = None
+    # outputConfig can be None, a str (eg 'screen') a dict (eg {'format': 'table'}) or a list (eg [{'type': 'screen', 'format': 'simple'}])
+    if outputConfig is None:
+        log.debug("got blank outputConfig")
+        outputType = DEFAULT_OUTPUT
+        _format = getFormatfromConfig(DEFAULT_FORMAT, device, topic)
+        log.debug("got format: %s" % (_format))
+        _output = getOutputClass(outputType, _format)
+        log.debug("got output: %s" % (_output))
+        return _output
+    elif isinstance(outputConfig, str):
+        # eg 'screen'
+        log.debug("got str type outputConfig: %s" % (outputConfig))
+        outputType = outputConfig
+        _format = getFormatfromConfig(DEFAULT_FORMAT, device, topic)
+        log.debug("got format: %s" % (_format))
+        _output = getOutputClass(outputType, _format)
+        log.debug("got output: %s" % (_output))
+        return _output
+    elif isinstance(outputConfig, dict):
+        # eg {'format': 'table'}
+        log.debug("got dict type outputConfig: %s" % (outputConfig))
+        outputType = outputConfig.get("type", DEFAULT_OUTPUT)
+        formatConfig = outputConfig.get("format", DEFAULT_FORMAT)
+        topic_override = outputConfig.get("topic_override")
+        # use topic override from the config
+        if topic_override is not None:
+            topic = topic_override
+        _format = getFormatfromConfig(formatConfig, device, topic)
+        log.debug("got format: %s" % (_format))
+        _output = getOutputClass(outputType, _format)
+        log.debug("got output: %s" % (_output))
+        return _output
+    elif isinstance(outputConfig, list):
+        # eg [{'type': 'screen', 'format': 'simple'}], possibly multiple outputs
+        # loop through outputs
+        log.debug("got list type outputConfig: %s" % (outputConfig))
+        outputs = []
+        for item in outputConfig:
+            _out = parseOutputConfig(item, topic, schedule_name, device, mqtt_broker)
+            if _out is not None:
+                outputs.append(_out)
+        return outputs
```

Comment:
A method shouldn't be able to return a single object or a list. It will make using it difficult. If we can return a list we should only return lists and set it as the return type.

Reply:
true - good catch